cron/internal/worker/main.go

```diff
@@ -143,10 +174,14 @@ func processRequest(ctx context.Context,
 	// TODO: run Scorecard for each repo in a separate thread.
 	for _, repoReq := range batchRequest.GetRepos() {
 		logger.Info(fmt.Sprintf("Running Scorecard for repo: %s", *repoReq.Url))
-		repo, err := githubrepo.MakeGithubRepo(*repoReq.Url)
-		if err != nil {
+		var repo clients.Repo
+		var err error
+		repoClient := githubClient
+		if repo, err = gitlabrepo.MakeGitlabRepo(*repoReq.Url); err != nil {
+			repoClient = gitlabClient
```

Comment:
same question about this making a gitlab request in the cron, which will result in millions of queries.

Reply:
API queries for client detection shouldn't happen unless the URL is not `github.com|gitlab.com`.